### PR TITLE
[SMALLFIX] improve Authority regrex and add tests

### PR DESCRIPTION
**Pull request by LuQQiu**
Change the SingleMasterAuthority and ZookeeperAuthority to be more concrete check.
add integration tests for using Alluxio on Zookeeper Uri in HDFS API to connect to Alluxio on Zookeeper cluster.

### CODE REVIEW

**Comment by LuQQiu** [comment]
jenkins, test this please

File: core/common/src/main/java/alluxio/uri/Authority.java [hunk 1]
```diff
@@ -23,7 +23,8 @@
  */
 public interface Authority extends Comparable<Authority>, Serializable {
   Logger LOG = LoggerFactory.getLogger(Authority.class);
-  Pattern ZOOKEEPER_AUTH = Pattern.compile("^zk@(.*)");
+  Pattern SINGLE_MASTER_AUTH = Pattern.compile("^([^:,;]+):(\\d+)$");
```
Comment (gpang) [inline core/common/src/main/java/alluxio/uri/Authority.java]: Did this fix a bug? Is a test case added for this?

File: core/common/src/test/java/alluxio/uri/AuthorityTest.java [hunk 2]
```diff
@@ -24,9 +24,10 @@
   @Test
   public void authorityFromStringTest() {
     assertTrue(Authority.fromString("localhost:19998") instanceof SingleMasterAuthority);
+    assertTrue(Authority.fromString("127.0.0.1:19998") instanceof SingleMasterAuthority);
```
Comment (gpang) [inline core/common/src/test/java/alluxio/uri/AuthorityTest.java]: Is this what failed before this fix?

File: core/common/src/main/java/alluxio/uri/Authority.java (hunk 1 above)
Comment (LuQQiu) [reply core/common/src/main/java/alluxio/uri/Authority.java]: This does not fix a bug, it just makes the check be more concrete and accurate.
In the previous way, we do not check if the port is an integer.

Some unit tests have been added in the `AuthorityTest`.

```
assertTrue(Authority.fromString("localhost:") instanceof UnknownAuthority);
assertTrue(Authority.fromString("localhost:asdsad") instanceof UnknownAuthority);
```
Those will be passed in the previous way as a SingleMasterAuthority but not in the new way.

File: core/common/src/test/java/alluxio/uri/AuthorityTest.java (hunk 2 above)
Comment (LuQQiu) [reply core/common/src/test/java/alluxio/uri/AuthorityTest.java]: This will be passed before the fix, I just add more cases of hostname here.

**Pull request merged**